plugins/catalyst/python/src/mqt/catalyst/__init__.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+# Copyright (c) 2025 Munich Quantum Software Company GmbH
+# All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+#
+# Licensed under the MIT License
+
+"""MQT Catalyst Plugin."""
+
+from __future__ import annotations
+
+from importlib.metadata import PackageNotFoundError, distribution
+from pathlib import Path
+
+import pennylane as qml
+
+from catalyst.passes import PassPlugin
+
+
+def get_catalyst_plugin_abs_path() -> Path:
+    """Locate the mqt-catalyst-plugin library.
+
+    Returns:
+        The absolute path to the mqt-catalyst-plugin library.
+
+    Raises:
+        FileNotFoundError: If the include directory is not found.
+        ImportError: If mqt-catalyst-plugin is not installed.
+    """
+    try:
+        dist = distribution("mqt-catalyst-plugin")
+        # Check for the plugin in the mqt-core package on Linux
+        catalyst_plugin_abs_path = Path(dist.locate_file("mqt/catalyst/lib/mqt-catalyst-plugin.so"))
+        if catalyst_plugin_abs_path.exists() and catalyst_plugin_abs_path.is_file():
+            return catalyst_plugin_abs_path
+        # Check for the plugin in the mqt-core package on macOS
+        catalyst_plugin_abs_path = Path(dist.locate_file("mqt/catalyst/lib/mqt-catalyst-plugin.dylib"))
+        if catalyst_plugin_abs_path.exists() and catalyst_plugin_abs_path.is_file():
+            return catalyst_plugin_abs_path
+        msg = "mqt-catalyst-plugin library not found."
+        raise FileNotFoundError(msg)
+    except PackageNotFoundError:
+        msg = "mqt-catalyst-plugin not installed, installation required to access the include files."
+        raise ImportError(msg) from None
```

Comment:
Depending on the answer to the other question on whether we actually need the `commands` file, this is definitely functionality that should be part of that module.